ocaml/utils/misc.ml

```diff
@@ -1099,6 +1099,29 @@ let output_of_print print =
   in
   output
 
+exception LimitExceeded
```

Comment:
Consider declaring this exception within `is_print_bigger_than`, to document the fact that it's only used within that function. I'm not concerned about creating a new extensible constructor each time the function is called. You could alternatively do something like `let is_print_bigger_than = let exception Limit_exceeded in fun size p -> ...` but it's a bit less natural.

Also please name it `Limit_exceeded`.